From 6a63a5e7c4c11275a3621b70d02727ceb9be9d3f Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Sun, 12 Aug 2018 23:12:21 -0700
Subject: [PATCH] lineagehw: Convert to Android.bp and use only common classes

*) Restoration of device specific lineagehw capabilities
   will happen later via hidl implementations.

Change-Id: Ib88f9ca74eb389936fe0096b010f9dce90c559d5
---
 Android.bp | 26 ++++++++++++++++++++++++++
 Android.mk | 53 -----------------------------------------------------
 2 files changed, 26 insertions(+), 53 deletions(-)
 create mode 100644 Android.bp
 delete mode 100644 Android.mk

diff --git a/Android.bp b/Android.bp
new file mode 100644
index 0000000..dd27f2e
--- /dev/null
+++ b/Android.bp
@@ -0,0 +1,26 @@
+//
+// Copyright (C) 2018 The LineageOS Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+java_library {
+    name: "org.lineageos.hardware",
+    srcs: ["src//**/*.java"],
+    static_libs: ["vendor.lineage.livedisplay-V1.0-java"],
+
+    libs: [
+        "org.lineageos.platform.internal",
+        "services",
+    ],
+}
diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 463f505..0000000
--- a/Android.mk
+++ /dev/null
@@ -1,53 +0,0 @@
-# Copyright (C) 2013 The CyanogenMod Project
-# Copyright (C) 2017-2018 The LineageOS Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-LOCAL_PATH := $(call my-dir)
-
-include $(CLEAR_VARS)
-
-ifneq ($(BOARD_HARDWARE_CLASS),)
-    $(foreach bcp, $(BOARD_HARDWARE_CLASS), \
-        $(eval BOARD_SRC_FILES += $(call all-java-files-under, ../../../$(bcp))))
-endif
-
-BASE_SRC_FILES += $(call all-java-files-under, src/)
-
-reverse = $(if $(wordlist 2,2,$(1)),$(call reverse,$(wordlist 2,$(words $(1)),$(1))) $(firstword $(1)),$(1))
-
-overriden_classes :=
-    $(foreach cf, $(call reverse, $(BOARD_SRC_FILES)), \
-        $(eval overriden_classes += $(cf)))
-
-unique_specific_classes :=
-    $(foreach cf, $(overriden_classes), \
-        $(if $(filter $(notdir $(unique_specific_classes)), $(notdir $(cf))),, \
-            $(eval unique_specific_classes += $(cf))))
-
-default_classes :=
-    $(foreach cf, $(BASE_SRC_FILES), \
-        $(if $(filter $(notdir $(unique_specific_classes)), $(notdir $(cf))),, \
-            $(eval default_classes += $(cf))))
-
-LOCAL_SRC_FILES += $(default_classes) $(unique_specific_classes)
-
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE := org.lineageos.hardware
-LOCAL_JAVA_LIBRARIES := org.lineageos.platform.internal services
-LOCAL_STATIC_JAVA_LIBRARIES := vendor.lineage.livedisplay-V1.0-java-static
-
-include $(BUILD_JAVA_LIBRARY)
-
-include $(call all-makefiles-under,$(LOCAL_PATH))
-
-- 
2.17.1

